### ISSUE TITLE
Deprecate kubernetes-ro service

### ISSUE THREAD

**Issue by erictune** [GoogleCloudPlatform/kubernetes#4567]
Now that the "secrets" type has been merged, and we have namespaces, I don't think it often makes sense to give blanket readonly access to objects.  So, lets deprecated it.

I don't know that we want to rip it out all in one go.
Nothing is using the `kubernetes-ro` service as far as I can see from searching the codebase, however some customers might depend on it, and maybe openshift does.  For GKE, I think we'd like it gone.  So, lets add a flag to the master that disables its creation.  If a customer needs it we can tell them how to get it back, but it will discourage new uses.
 
Once service-accounts land, we can give people that need it a way to automatically generate policy and credentials for things in pods.  Then we can delete `kubernetes-ro` it altogether.

**Comment by derekwaynecarr** [comment]
OpenShift is most interested in the kubernetes service.  Is that going away on GKE?  I actually plan to use that to support running OpenShift as a pod on Kubernetes.

**Comment by lavalamp** [comment]
I think it is used by third parties-- isn't that the way our SkyDNS pod knows where to get the list of things to populate DNS with?

**Comment by vmarmol** [comment]
Heapster uses the RO service to get information about running things

**Comment by thockin** [comment]
DNS uses it.  I sort of like that it is automatically rate limited, but I
also have not looked at the secrets stuff yet.

I will go with the flow, but please don't break DNS.

**Comment by erictune** [comment]
@derekwaynecarr @lavalamp @thockin @vmarmol 
We aren't going to break the existing use cases.  They will be grandfathered in until we have the ability to automatically distribute credentials to pods.  Then each existing use case will be lovingly and tenderly migrated to the `kubernetes` service.

@thockin 
yes rate limiting is nice.  we should have it for the `kubernetes` service.

**Comment by erictune** [comment]
@derekwaynecarr 
no, GKE does not want to get rid of `kubernetes` service.  It wants to get rid of the insecure endpoint on the apiserver.

**Comment by erictune** [comment]
#5921

**Comment by erictune** [comment]
#5917

**Comment by davidopp** [comment]
@erictune can you confirm that deprecating kubernetes-ro is needed for v1.0 (as opposed to the other issues you mentioned, and this one being optional for 1.0)?

**Comment by erictune** [comment]
This is not needed for 1.0, but the other two are.  Fixing labels.

**Comment by erictune** [comment]
thinking about this more, this is needed for v1.

**Comment by davidopp** [comment]
@erictune Is it OK if we assign this issue to you?

**Comment by erictune** [comment]
Yes.

On Tue, Apr 28, 2015 at 2:54 PM, David Oppenheimer <notifications@github.com

**Comment by davidopp** [comment]
#7444 and #5921 are vaguely related and both marked for 1.0

**Comment by sujaymansingh** [comment]
This is a rookie question, but if the RO service is being removed, how would I access the list of services from within a container?

**Comment by roberthbailey** [comment]
Use a service account to inject credentials and access the secure (read/write) port on the master.

**Comment by erictune** [comment]
@ sujaymansingh  if you describe your client, I can suggest the best way to
access the API.

**Comment by sujaymansingh** [comment]
Hi @erictune 
Sorry about the delay and thanks for reaching out!

It will mainly be python (though if it is HTTP anything could access it right?)

I'm mainly interested in a mix of python APIs (probably using flask) and standalone scripts.
